File: examples/agent-sdk/typescript/basic_quality_check.ts
// basic_quality_check.ts
// Prerequisites: npm install @anthropic-ai/claude-code
// ANTHROPIC_API_KEY and TAPPS_MCP_PROJECT_ROOT must be set

import { query, type ClaudeCodeOptions } from "@anthropic-ai/claude-code";

async function runQualityCheck(filePath: string): Promise<void> {
  const options: ClaudeCodeOptions = {
    // Configure TappsMCP as an MCP server
    mcpServers: {
      "tapps-mcp": {
        command: "uvx",
        args: ["tapps-mcp", "serve"],
        env: {
          TAPPS_MCP_PROJECT_ROOT:
            process.env.TAPPS_MCP_PROJECT_ROOT ?? process.cwd(),
        },
      },
    },
    // Restrict tool access to only the quick check tool
    allowedTools: ["mcp__tapps-mcp__tapps_quick_check"],
    maxTurns: 3,
  };

  for await (const message of query({
    prompt: `Run tapps_quick_check on ${filePath} and report the score and top issues.`,
    options,
  })) {
    if ("content" in message) {
      console.log(message.content);
    }
  }
}

const filePath = process.argv[2] ?? "src/main.py";
runQualityCheck(filePath).catch(console.error);

File: examples/agent-sdk/python/basic_quality_check.py
#!/usr/bin/env python3
"""Basic TappsMCP + Agent SDK example.

Runs a quick quality check on a single file using the Claude Agent SDK
with TappsMCP configured as an MCP server.

Prerequisites:
  pip install claude-code-sdk
  export ANTHROPIC_API_KEY=sk-...
  export TAPPS_MCP_PROJECT_ROOT=/path/to/your/project
"""
from __future__ import annotations

import asyncio
import os
import sys

from claude_code_sdk import ClaudeCodeOptions, McpServerConfig, query


async def run_quality_check(file_path: str) -> str:
    """Run a TappsMCP quick check on a single file via the Agent SDK."""
    # Configure TappsMCP as an MCP server
    mcp_servers = {
        "tapps-mcp": McpServerConfig(
            command="uvx",
            args=["tapps-mcp", "serve"],
            env={
                "TAPPS_MCP_PROJECT_ROOT": os.environ.get(
                    "TAPPS_MCP_PROJECT_ROOT",
                    os.getcwd(),
                ),
            },
        ),
    }

    # Create options with TappsMCP and restricted tool access
    options = ClaudeCodeOptions(
        mcp_servers=mcp_servers,
        allowed_tools=["mcp__tapps-mcp__tapps_quick_check"],
        max_turns=3,
    )

    # Run the query and collect results
    result_parts: list[str] = []
    async for message in query(
        prompt=(
            f"Run tapps_quick_check on {file_path} and report "
            "the score and top issues."
        ),
        options=options,
    ):
        if hasattr(message, "content"):
            result_parts.append(str(message.content))

    return "\n".join(result_parts)


if __name__ == "__main__":
    file_to_check = sys.argv[1] if len(sys.argv) > 1 else "src/main.py"
    result = asyncio.run(run_quality_check(file_to_check))
    print(result)
